### ISSUE TITLE
[BUG] Improve  Error Message for 'Missing Spawn Location Point' 

### ISSUE THREAD

**Issue by adunis** [Kruptein/PlanarAlly#735]
I suggest adding to the Error Message for the Missing Spawn Point how to do that if a DM doesn't know what it is. 
For example "You can create a Spawn Location point in the DM layer by left clicking on the map". 

![image](https://user-images.githubusercontent.com/47754845/115509557-dfd83680-a27e-11eb-9540-e84b412e2f02.png)

**Comment by dthv** [comment]
@adunis Well, I just checked and I create Spawn Points with a right-click. There would have to be a check whether mouse buttons are inverted in order to not send users in the wrong direction.

Maybe it is easier to auto-create a spawn point on location generation? Finally, spawn points are [covered extensively in the documentation](https://www.planarally.io/docs/dm/locations/#moving-shapes-to-another-location).

**Comment by Kruptein** [comment]
> Maybe it is easier to auto-create a spawn point on location generation?
Not a bad first instinct, but this was the original behaviour and opened up a lot of annoyances, so I would rather not reintroduce that

**Comment by dthv** [comment]
I honestly would not like to see a "see the documentation"-info popping up everywhere. I strongly believe in `rtfm`. ;)

**Comment by dthv** [comment]
Well, I'm preaching the teachings of RTFM where I can – to the extent that I think it is not necessary to explicityl reference documentations.

My major concern with @adunis' proposal was that it referenced a *left click* where, by default, users use a *right click*. The operating systems I know allow to flip mouse buttons (e.g. for left-handed people), in this case a hint to "left/right" clicks might be confusing.

Thus, I'd go for "create at least one spawn point via the context-menu on the DM layer". I'd expect people to try around and do the occasional context-menu-click on the DM layer so this should be help enough.

Explicitly referencing the docs in this context would indicate that this is especially difficult and not intuitive enough to let users find out. One could, of course, do this, yet would have to expand every error message with "hey, there's documentation" to be consistent. I would not like this. When people use new software and encounter a problem, their first instinct should lead them to the docs. There's nothing helpful in shoving that into their faces. But that's just kinda my opinion. :slightly_smiling_face:

Anway, **another approach** would be to, on creation of a new location:
1. move DM to DM layer
2. open a popup saying "make sure to create at least one spawn location via the context menu, otherwise tokens cannot be moved here"

**Just another approach** would be to default spawning at center if there's no spawn-point present, maybe with a popup saying "tokens were moved to center of Location xyz, you can spawn them at other places by creating a spawn point via the context menu on the DM layer", maybe with the possibility of "don't show this again".

Then again, since this should indeed be something of showing once, but never again, 'cause then you learned it', links to docs should be fine…

**Comment by dthv** [comment]
…aaaand just another point to maybe make it a bit more clear: It's about accessing the context menu on a specific layer, the rest should be self-explaining. That could be included into a message. It's not necessary to show users to the docs for something that easy, I think. It's more work where less would suffice.

**Comment by Kruptein** [comment]
It's actually not necessary to move to the DM layer to get the entry in the context menu.  You can create a spawn location from any location, it's just that they'll always be created on the DM layer

But I agree with your assessment that adding links to the documentation everywhere might become overkill/noisy.

The example sentence you used is I think a good addition, I'll go for that one.
> "create at least one spawn point via the context-menu on the DM layer".

**Comment by dthv** [comment]
> adding links to the documentation everywhere might become overkill/noisy

Yeah, that's what I wanted to express but failed to write, properly. :slightly_smiling_face:
